python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/ethernet-interfaces.j2

```diff
@@ -160,6 +160,47 @@ interface {{ ethernet_interface.name }}
 {%     if ethernet_interface.pvlan_mapping is arista.avd.defined %}
    switchport pvlan mapping {{ ethernet_interface.pvlan_mapping }}
 {%     endif %}
+{%     if ethernet_interface.vlan_id is arista.avd.defined and ethernet_interface.type | arista.avd.default != 'l2dot1q' %}
+   vlan id {{ ethernet_interface.vlan_id }}
+{%     endif %}
+{%     if ethernet_interface.encapsulation_dot1q.vlan is arista.avd.defined %}
+{%         set encapsulation_dot1q_cli = 'encapsulation dot1q vlan ' ~ ethernet_interface.encapsulation_dot1q.vlan %}
+{%         if ethernet_interface.encapsulation_dot1q.inner_vlan is arista.avd.defined %}
+{%             set encapsulation_dot1q_cli = encapsulation_dot1q_cli ~ ' inner ' ~ ethernet_interface.encapsulation_dot1q.inner_vlan %}
+{%         endif %}
+   {{ encapsulation_dot1q_cli }}
+{%     elif ethernet_interface.encapsulation_vlan.client.protocol is arista.avd.defined %}
+{%         set protocol = ethernet_interface.encapsulation_vlan.client.protocol %}
+{%         set network_flag = False %}
+{%         if protocol in ['dot1q', 'dot1ad'] %}
+{%             if ethernet_interface.encapsulation_vlan.client.vlan is arista.avd.defined %}
+{%                 set encapsulation_cli = 'client ' ~ protocol ~ ' ' ~ ethernet_interface.encapsulation_vlan.client.vlan %}
+{%             elif ethernet_interface.encapsulation_vlan.client.outer is arista.avd.defined and ethernet_interface.encapsulation_vlan.client.inner is arista.avd.defined %}
+{%                 set encapsulation_cli = 'client ' ~ protocol ~ ' outer ' ~ ethernet_interface.encapsulation_vlan.client.outer ~ ' inner ' ~ ethernet_interface.encapsulation_vlan.client.inner %}
+{%                 if ethernet_interface.encapsulation_vlan.network.protocol | arista.avd.default == 'client inner' %}
+{%                     set network_flag = True %}
+{%                     set encapsulation_cli = encapsulation_cli ~ ' network ' ~ ethernet_interface.encapsulation_vlan.network.protocol %}
+{%                 endif %}
+{%             endif %}
+{%         elif protocol in ['untagged', 'unmatched'] %}
+{%             set encapsulation_cli = 'client ' ~ ethernet_interface.encapsulation_vlan.client.protocol %}
+{%         endif %}
+{%         if encapsulation_cli is arista.avd.defined %}
+{%             if protocol in ['dot1q', 'dot1ad', 'untagged'] and ethernet_interface.encapsulation_vlan.network.protocol is arista.avd.defined and not network_flag %}
+{%                 if ethernet_interface.encapsulation_vlan.network.vlan is arista.avd.defined %}
+{%                     set encapsulation_cli = encapsulation_cli ~ ' network ' ~ ethernet_interface.encapsulation_vlan.network.protocol ~ ' ' ~ ethernet_interface.encapsulation_vlan.network.vlan %}
```

Comment:
set the network protocol as you done for client protocol ethernet_interface.encapsulation_vlan.network.protocol

Reply:
Fixed